analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -81,6 +86,24 @@ public ProjectConfigReader ProjectConfiguration()
         }
     }
 
+    /// <summary>
+    /// Reads the properties from the SonarLint.xml file and caches the result for the scope of this analysis.
+    /// </summary>
+    public SonarLintXmlReader SonarLintFile()
+    {
+        if (Options.SonarLintXml() is { } sonarLintXml)
+        {
+            return sonarLintXml.GetText() is { } sourceText
+                && AnalysisContext.TryGetValue(sourceText, GetSonarLintXml(Compilation.Language), out var cachedSonarLintXmlReader)
```

Comment:
It is not always the cahced one, if it does not exist it will be created and put to cache.
```suggestion
                && AnalysisContext.TryGetValue(sourceText, GetSonarLintXml(Compilation.Language), out var sonarLintXmlReader)
```